src/server/query_processor.py

```diff
@@ -71,10 +271,22 @@ async def process_query(
         include_patterns=pattern if pattern_type == PatternType.INCLUDE else None,
     )
 
+    # Check if digest already exists on S3 before cloning
+    s3_response = await _check_s3_cache(
+        query=query,
+        input_text=input_text,
+        slider_position=slider_position,
```

Comment:
I think you want to deal with computed file size here instead of slider_position